packages/theme/src/stylelint-plugins/no-unknown-ds-tokens.mjs

```diff
@@ -0,0 +1,101 @@
+import stylelint from 'stylelint';
+import tokenList from '../prebuilt/js/design-tokens.js';
+
+const DS_TOKEN_PREFIX = 'wpds-';
+
+/**
+ * Extracts all unique CSS custom properties (variables) from a given CSS value string,
+ * including those in fallback positions, optionally filtering by a specific prefix.
+ *
+ * @param {string} value       - The CSS value string to search for variables.
+ * @param {string} [prefix=''] - Optional prefix to filter variables (e.g., 'wpds-').
+ * @return {Set<string>} A Set of unique matched CSS variable names (e.g., Set { '--wpds-token' }).
+ *
+ * @example
+ * extractCSSVariables(
+ *   'border: 1px solid var(--wpds-border-color, var(--wpds-border-fallback)); ' +
+ *   'color: var(--wpds-color-fg, black); ' +
+ *   'background: var(--unrelated-bg);',
+ *   'wpds'
+ * );
+ * // → Set { '--wpds-border-color', '--wpds-border-fallback', '--wpds-color-fg' }
+ */
+function extractCSSVariables( value, prefix = '' ) {
+	const regex = /--[\w-]+/g;
+	const variables = new Set();
+
+	let match;
+	while ( ( match = regex.exec( value ) ) !== null ) {
+		const variableName = match[ 0 ];
+		if ( variableName.startsWith( `--${ prefix }` ) ) {
+			variables.add( variableName );
+		}
+	}
+
+	return variables;
+}
+
+const knownTokens = new Set( tokenList );
+const wpdsTokensRegex = new RegExp( `[^\\w]--${ DS_TOKEN_PREFIX }`, 'i' );
+
+const {
+	createPlugin,
+	utils: { report, ruleMessages, validateOptions },
+} = stylelint;
+
+const ruleName = 'plugin-wpds/no-unknown-ds-tokens';
+
+const messages = ruleMessages( ruleName, {
+	rejected: ( tokenNames ) =>
+		`The following CSS variables are not valid (public) Design System tokens: ${ tokenNames }`,
```

Comment:
@ciampo What do you think about this wording? Too long? I felt the "(public)" might be a bit confusing without context.

```suggestion
		`The following CSS variables are not valid Design System tokens, or are private tokens that should not be used directly: ${ tokenNames }`,
```

Reply:
You're right! Probably no need to mention the private part. Probably a change that was applied while I was away?